packages/flutter/lib/src/material/material_button.dart

```diff
@@ -0,0 +1,298 @@
+  final Widget child;
+
+  /// Whether the button is enabled or disabled.
+  ///
+  /// Buttons are disabled by default. To enable a button, set its [onPressed]
+  /// property to a non-null value.
+  bool get enabled => onPressed != null;
+
+  /// The internal padding for the button's [child].
+  ///
+  /// Defaults to the value from the current [ButtonTheme],
+  /// [ButtonThemeData.padding].
+  final EdgeInsetsGeometry padding;
+
+  /// The shape of the button's [Material].
+  ///
+  /// The button's highlight and splash are clipped to this shape. If the
+  /// button has an elevation, then its drop shadow is defined by this
+  /// shape as well.
+  final ShapeBorder shape;
+
+  /// {@macro flutter.widgets.Clip}
+  final Clip clipBehavior;
+
+  /// Defines the duration of animated changes for [shape] and [elevation].
+  ///
+  /// The default value is [kThemeChangeDuration].
+  final Duration animationDuration;
+
+  /// Configures the minimum size of the tap target.
+  ///
+  /// Defaults to [ThemeData.materialTapTargetSize].
+  ///
+  /// See also:
+  ///
+  ///   * [MaterialTapTargetSize], for a description of how this affects tap targets.
+  final MaterialTapTargetSize materialTapTargetSize;
+
+  /// The smallest horizontal extent that the button will occupy.
+  ///
+  /// Defaults to the value from the current [ButtonTheme].
+  final double minWidth;
+
+  /// The vertical extent of the button.
+  ///
+  /// Defaults to the value from the current [ButtonTheme].
+  final double height;
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final ButtonThemeData buttonTheme = ButtonTheme.of(context);
+
+    return RawMaterialButton(
+      onPressed: onPressed,
+      fillColor: color,
+      textStyle: theme.textTheme.button.copyWith(color: buttonTheme.getTextColor(this)),
+      highlightColor: highlightColor ?? theme.highlightColor,
+      splashColor: splashColor ?? theme.splashColor,
+      elevation: buttonTheme.getElevation(this),
+      highlightElevation: buttonTheme.getHighlightElevation(this),
+      padding: buttonTheme.getPadding(this),
+      constraints: buttonTheme.constraints,
+      shape: buttonTheme.shape,
+      clipBehavior: clipBehavior ?? Clip.none,
+      animationDuration: buttonTheme.getAnimationDuration(this),
+      child: child,
+      materialTapTargetSize: materialTapTargetSize ?? theme.materialTapTargetSize,
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(FlagProperty('enabled', value: enabled, ifFalse: 'disabled'));
+  }
+}
+
+/// The type of [MaterialButton]s created with [RaisedButton.icon], [FlatButton.icon],
+/// and [OutlineButton.icon].
+//
+// This class only exists to give the label and icon button widgets a distinct
+/// class for the sake of [ButtonTheme]. It can not be instantiated.
+abstract class MaterialButtonWithIconMixin {
```

Comment:
I believe you can use the new mixin syntax now: `mixin MaterialButtonWithIconMixin { }` right @yjbanov ?

Reply:
The mixin syntax appears to work.

Reply:
The analyzer and the linter still produce warnings on some valid code, but if you are not triggering those warnings, you should be OK.